core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/PauseManager.scala

```diff
@@ -15,6 +15,8 @@ class PauseManager {
   // current pause privilege level
   private val pausePrivilegeLevel = new AtomicInteger(PauseManager.NoPause)
   // yielded control of the dp thread
+  // volatile is necessary otherwise main thread cannot notice the change
```

Comment:
please explain this comment a bit more

Reply:
@shengquan-ni read some online docs and `volatile` means read/writes are through memory rather than CPU cache. It makes me wondering why there's no issue on the prototype without the volatile keyword. But I think it's good to add it to be same. Furthermore, I believe all varaibles that are shared between main actor thread and dp thread will need this annotation. Do we want to add the annotations later?